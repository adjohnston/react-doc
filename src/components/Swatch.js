import React from 'react'
import PropTypes from 'prop-types'
import className from 'helpers/class-name'

const Swatch = ({swatch, onClick}) => {
  const handleOnClick = () => {
    return onClick(swatch)
  }

  return (
    <button
      className={className('swatch-button')}
      style={{background: swatch}}
      onClick={handleOnClick} />
  )
}

Swatch.propTypes = {
  onClick: PropTypes.func.isRequired,
  swatch: PropTypes.string
}

export default Swatch
