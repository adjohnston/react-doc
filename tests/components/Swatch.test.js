import React from 'react'
import renderer from 'react-test-renderer'
import Swatch from 'components/Swatch'

test('it renders correctly', () => {
  const props = {
    swatch: 'black',
    onClick: () => {}
  }

  const tree = renderer
    .create(<Swatch {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
