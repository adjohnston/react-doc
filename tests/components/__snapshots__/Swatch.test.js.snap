// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<button
  className="stncl__swatch-button"
  onClick={[Function]}
  style={
    Object {
      "background": "black",
    }
  }
/>
`;
